enhancements/oc/cli-manager.md

```diff
@@ -0,0 +1,252 @@
+* `oc tools install odo` will install `odo` to a user's home directory, as a standalone binary.
+
+##### Example
+```text
+$ oc tools list
+NAME                 DESC                        LATEST       INSTALLED
+-----                -----                       -----        -----
+kubectl              Kubernetes CLI              1.15         1.13
+oc                   OpenShift CLI               4.4          4.3
+odo                  OpenShift Developer CLI     1.0          Not Installed
+kn                   Knative CLI                 1.0          Not Installed
+tkn                  Tekton CLI                  1.0          Not Installed
+
+```
+
+#### Story 2
+
+As a user, I want to access various CLIs from the OpenShift Console Web Terminal so that I can use the
+services available in the cluster. The CLIs versions should match the those of the operators deployed
+on the cluster.
+
+Currently, users can access `consoleclidownloads` custom resources for helm, oc, and odo either from oc with `oc get consoleclidownloads` or through
+the `downloads -n openshift-console` route. This access will still be available, through the updated ClusterCLI custom resources. The ClusterCLI custom
+resources can also be accessed through the console by searching for `ClusterCLI` as you would any other resource. The CLI versions that are compatible
+with a given release payload will in this way be available to the Console Web Terminal. CLI artifacts will continue to be served via a webserver for users to download
+from a route as they are today with the console downloads route. Today, only oc binaries are available with the console downloads route. The other consolclidownloads
+custom resources offer the external links to the user. Other CLIs can provide an artifact image to be shipped with the release payload if they wish to offer binaries
+served in this way from an imagestream located within the cluster registry. The CLIS currently available as `consoleclidownloads`, `oc, helm, and odo`, will be
+included in the reference implementation of this enhancement.
+
+The deployment of the CRD controller will have an API accessible to the console for future CLI list and download integration in the future.
+
+#### Story 3
+
+As owner of a CLI tool, I want to publish my CLI tool to users of the cluster. I need to create a CR for my tool, or provide the required information about my tool
+to a cluster-admin for the creation of a CR:
+* Name
+* Short description
+* Long description
+* Caveats
+* Homepage
+* Version
+* Platform/architecture
+* The image:tag (and registry credentials if required)
+* The path within the image where the binary for the given platform/architecture can be found
+
+### Risks and Mitigations
+
+Distributing undesirable binaries is always a risk. Some mitigations include requiring cluster-admins to maintain the index, and the verification of downloaded
+binaries using SHA256 hashes.
```

Comment:
Verifying SHA256 hashes is only good if these hashes can be trusted (for example because they are signed with a RH-owned key). Might be useful to clarify this if this can be done in a few words.

Reply:
I've updated to hopefully clarify that it's the responsibility of the cluster-admins to publish trusted binaries.